src/main/java/org/openrewrite/staticanalysis/RemoveToStringCallsFromArrayInstances.java

```diff
@@ -0,0 +1,132 @@
+    private static final List<String> PATTERNS = Arrays.asList(
+            "java.io.PrintStream print*(Object)",
+            "java.lang.String format*(..)",
+            "java.lang.String valueOf(java.lang.Object)",
+            "java.lang.StringBuilder insert(int, Object)",
+            "java.lang.StringBuilder append(Object)",
+            "java.io.PrintStream format(String, Object[])"
+    );
+    private static final List<MethodMatcher> METHOD_MATCHERS = PATTERNS.stream().map(MethodMatcher::new).collect(Collectors.toList());
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-2116");
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Remove `toString()` calls on arrays";
+    }
+
+    @Override
+    public String getDescription() {
+        return "The result from `toString()` calls on arrays is largely useless. The output does not actually reflect" +
+               " the contents of the array. `Arrays.toString(array)` give the contents of the array.";
+    }
+
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveToStringFromArraysVisitor();
+    }
+
+    private static class RemoveToStringFromArraysVisitor extends JavaVisitor<ExecutionContext> {
+        @Override
+        public J visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
+            if (OBJECT_TOSTRING_MATCHER.matches(mi) || OBJECTS_TOSTRING_MATCHER.matches(mi)) {
+                Expression select = mi.getSelect();
+
+                if (select != null) {
+                    if (!(select.getType() instanceof JavaType.Array)) {
+                        return mi;
+                    }
+
+                    maybeAddImport("java.util.Arrays");
+                    return JavaTemplate.builder("Arrays.toString(#{anyArray(java.lang.Object)})")
+                            .imports("java.util.Arrays")
+                            .build()
+                            .apply(getCursor(), mi.getCoordinates().replace(), select);
+                }
+            } else if (METHOD_MATCHERS.stream().anyMatch(matcher -> matcher.matches(mi))) {
+                // deals with edge cases where .toString() is called implicitly
+                List<Expression> arguments = mi.getArguments();
+                for (Expression arg : arguments) {
+                    if (arg.getType() instanceof JavaType.Array) {
+                        Cursor c = getCursor();
+                        c.putMessage("METHOD_KEY", mi);
+                        break;
+                    }
+                }
+            }
+
+            return super.visitMethodInvocation(mi, ctx);
+        }
+
+        @Override
+        public Expression visitExpression(Expression exp, ExecutionContext ctx) {
+            Expression e = (Expression) super.visitExpression(exp, ctx);
+            if (e.getType() instanceof JavaType.Array) {
```

Comment:
When you add a test like `System.out.println("" + (arr))` (where `arr` is an array), you will notice that you get a runtime exception. That is because `J.Parentheses` doesn't extend `TypedTree` (and I currently can't tell you if it should or not). To fix this my suggestion is to simply add an `instanceof` check:

```suggestion
            if (e instanceof TypedTree && e.getType() instanceof JavaType.Array) {
```